File: src/utils/storage.ts
import AsyncStorage from '@react-native-async-storage/async-storage';

const LANGUAGE_STORAGE_KEY = '@offsight:selected_language';

export interface Language {
  code: string;
  name: string;
}

export const SUPPORTED_LANGUAGES: Language[] = [
  { code: 'en', name: 'English' },
  { code: 'es', name: 'Spanish' },
  { code: 'fr', name: 'French' },
  { code: 'de', name: 'German' },
  { code: 'it', name: 'Italian' },
  { code: 'pt', name: 'Portuguese' },
  { code: 'ru', name: 'Russian' },
  { code: 'ja', name: 'Japanese' },
  { code: 'ko', name: 'Korean' },
  { code: 'zh', name: 'Chinese' },
  { code: 'ar', name: 'Arabic' },
  { code: 'hi', name: 'Hindi' },
];

export async function getSelectedLanguage(): Promise<string | null> {
  try {
    const language = await AsyncStorage.getItem(LANGUAGE_STORAGE_KEY);
    return language;
  } catch (error) {
    console.error('Error getting selected language:', error);
    return null;
  }
}

export async function saveSelectedLanguage(languageCode: string): Promise<void> {
  try {
    await AsyncStorage.setItem(LANGUAGE_STORAGE_KEY, languageCode);
  } catch (error) {
    console.error('Error saving selected language:', error);
  }
}

File: lib/utils/storage.dart
import 'package:shared_preferences/shared_preferences.dart';

const String languageStorageKey = '@offsight:selected_language';

class Language {
  final String code;
  final String name;

  const Language({
    required this.code,
    required this.name,
  });
}

const List<Language> supportedLanguages = [
  Language(code: 'en', name: 'English'),
  Language(code: 'es', name: 'Spanish'),
  Language(code: 'fr', name: 'French'),
  Language(code: 'de', name: 'German'),
  Language(code: 'it', name: 'Italian'),
  Language(code: 'pt', name: 'Portuguese'),
  Language(code: 'ru', name: 'Russian'),
  Language(code: 'ja', name: 'Japanese'),
  Language(code: 'ko', name: 'Korean'),
  Language(code: 'zh', name: 'Chinese'),
  Language(code: 'ar', name: 'Arabic'),
  Language(code: 'hi', name: 'Hindi'),
];

Future<String?> getSelectedLanguage() async {
  try {
    final prefs = await SharedPreferences.getInstance();
    return prefs.getString(languageStorageKey);
  } catch (error) {
    print('Error getting selected language: $error');
    return null;
  }
}

Future<void> saveSelectedLanguage(String languageCode) async {
  try {
    final prefs = await SharedPreferences.getInstance();
    await prefs.setString(languageStorageKey, languageCode);
  } catch (error) {
    print('Error saving selected language: $error');
  }
}

